File: DownloadFiles/TGNet/Chapter08/Chapter8DemoCode/C#/ADOTestDemo/ADODataReaderDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Data.SqlClient;
using System.Diagnostics;

namespace ADODataReaderDemo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            SqlConnection cn = new SqlConnection("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes");
            Int16 iRet;
            SqlCommand cmd = new SqlCommand("Select count(*) from customers", cn);
            const int iEXPECTED = 91;
            cn.Open();
            iRet = Convert.ToInt16 (cmd.ExecuteScalar());
            if (iRet == iEXPECTED)
            {
                MessageBox.Show("Test passed; Expected: " + iEXPECTED + " Actual: " + iRet);
            }
            else
            {
                MessageBox.Show("Test failed; Expected: " + iEXPECTED + " Actual: " + iRet);
            }
            cn.Close();

        }
    }
}

File: DownloadFiles/TGNet/Chapter08/Chapter8DemoCode/VB/ADOTestDemoSQLServer/ADOTestDemo/Form1.vb
Imports System.Data.sqlclient
'*************************************************************
'* Northwind Test. 
'* Verify that the expected number of records
'* in the Customers table matches the actual number of records.
'* Dependencies:
'* References: 
'*************************************************************** 


Public Class Form1

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        
        Dim cn As New SqlConnection("Data Source=localhost;Database=Northwind;Trusted_Connection=Yes")
        Dim iRet As Int16
        Dim cmd As New SqlCommand("Select count(*) from customers", cn)
        Const iEXPECTED As Integer = 91 'set number of expected items
        cn.Open()
        iRet = cmd.ExecuteScalar
        If iRet = iEXPECTED Then
            MessageBox.Show("Test passed; Expected: " & iEXPECTED & " Actual: " & iRet)
        Else
            MessageBox.Show("Test failed; Expected: " + iEXPECTED + " Actual: " + iRet)
        End If
        cn.Close()

    End Sub
End Class
